src/table/utils/styling-utils.js

```diff
@@ -25,58 +27,78 @@ export const SELECTION_STYLING = {
+  };
+  // Remove all Undefined Values from an Object
+  Object.keys(baseStyle).forEach((key) => baseStyle[key] == null && delete baseStyle[key]);
+  return baseStyle;
+};
 
 // Both index === -1 and color === null must be true for the property to be unset
 export const isUnset = (prop) => !prop || JSON.stringify(prop) === JSON.stringify({ index: -1, color: null });
 
 export function getHeadStyle(layout, theme) {
   const header = layout.components?.[0]?.header;
-  return header ? getBaseStyling(header, theme) : { padding: STYLING_DEFAULTS.PADDING };
+
+  return getBaseStyling(header, 'header', theme);
 }
 
 export function getBodyStyle(layout, theme) {
   const content = layout.components?.[0]?.content;
-  if (!content) return { padding: STYLING_DEFAULTS.PADDING };
 
   // Cases when hoverEffect is true:
   // 1. There is no hover font color but a hover background color,
-  // when hovering, the hover font color becomes white when the hover background color is a dark color
-  // or the hover font color stays the same as whetever the font color is when the hover background color is a light color.
+  // when hovering, the hover font color becomes white when the hover
+  // background color is a dark color or the hover font color stays
+  // the same as whatever the font color is when the hover background
+  // color is a light color.
   // 2. There is a hover font color but no hover background color,
-  // when hovering, only a hover font color is applied and the hover background color disappears.
-  // 3. There is no hover font color and no hover background color, when hovering, the defalut hover effect (light gray backgournd)
-  // 4. There are both hover font and background colors, when hovering, the hover font and background colors take effect.
+  // when hovering, only a hover font color is applied and the hover
+  // background color disappears.
+  // 3. There is no hover font color and no hover background color,
+  // when hovering, the default hover effect (light gray background)
+  // 4. There are both hover font and background colors, when hovering,
+  // the hover font and background colors take effect.
 
-  const unsetHoverBackgroundColor = isUnset(content.hoverColor);
-  const unsetHoverFontandBackgroundColor = isUnset(content.hoverFontColor) && unsetHoverBackgroundColor;
+  const unsetHoverBackgroundColor = isUnset(content?.hoverColor);
+  const unsetHoverFontBackgroundColor = isUnset(content?.hoverFontColor) && unsetHoverBackgroundColor;
 
-  const hoverBackgroundColor = unsetHoverFontandBackgroundColor
-    ? STYLING_DEFAULTS.HOVER_BACKGROUND
-    : unsetHoverBackgroundColor
+  const setHoverBackgroundColor = unsetHoverBackgroundColor
     ? ''
-    : getColor(content.hoverColor, STYLING_DEFAULTS.HOVER_BACKGROUND, theme);
-  const hoverFontColor = unsetHoverFontandBackgroundColor
+    : getColor(STYLING_DEFAULTS.HOVER_BACKGROUND, theme, content?.hoverColor);
```

Comment:
don't we know that `content.hoverColor` exists, since we checked with isUnset?

Reply:
> yes, we know that. So `content?.hoverColor` should be `content.hoverColor` here.

Reply:
Updated in #453

Reply:
this is also back

Reply:
because #453 is not merged yet